docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,332 @@
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t.typeText(developerNameInput, 'Peter');
+
+    //an awaited selector doesn't update and produces unstable test results. Avoid it.
+    const developerName = await Selector('#developer-name').value;
+    //
+
+    await t
+            .expect(developerName).eql('Peter')
+            .typeText(developerNameInput, 'Jack')
+            .expect(developerName).eql('Jack'); // fails
+});
+```
+
+In this snippet, the `developerName` is initialized with the value of a Selector, but because of the `await` keyword, the value is calculated once and doesn't update. This disables the smart assertions query mechanism, which in this case leads to inconclusive results.
+
+To avoid this, omit the `await` keyword:
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t.typeText(developerNameInput, 'Peter');
+
+    const developerName = Selector('#developer-name').value;
+
+    await t
+            .expect(developerName).eql('Peter')
+            .typeText(developerNameInput, 'Jack')
+            .expect(developerName).eql('Jack'); // passes
+});
+```
+
+The `developerName` is now initialized with a re-executable Selector API promise. When a test controller receives this promise, it enables the smart assertion query mechanism and TestCafe can wait for the value to update.
+
+> The smart assertion query mechanism works with Client Functions and promises created by TestCafe Selector API. Use the `await` keyword with user-created promises and promises returned from third-party libraries.
+
+This rule applies to ClientFunctions as well.
+
+The test below yields inconsistent results:
+
+```js
+import { Selector, ClientFunction } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with ClientFunction', async t => {
+    const getValue        = ClientFunction(() => document.getElementById('preferred-interface').value)
+    const interfaceSelect = Selector('#preferred-interface');
+    const interfaceOption = interfaceSelect.find('option');
+    const value           = await getValue();
+
+
+    await t
+        .click(interfaceSelect)
+        .click(interfaceOption.withText('JavaScript API'))
+        .expect(value).eql('JavaScript API')
+});
+```
+
+In this example, the client function obtains the value of a drop-down element. This value is then passed to the `t.expect` method. Since the client function is awaited when it's assigned to the variable, the value gets resolved instantly and never updates. The smart assertion query mechanism doesn't apply and the test fails.
```

Comment:
> the client function is awaited when it's assigned to the variable

1) the function is assigned to the variable? the function is not assigned (it's the result, Promise or value, depending on `await`), and not to the variable (it's a constant)

2) the correctors would say that 'function is awaited' is a slang

*Since you call `getValue()` with `await`,..*

Reply:
> Since the `getValue()` is called with `await`, the return value gets resolved instantly and never updates.

OR

> Since the  `await` keyword is used when calling `getValue()`, the return value gets resolved instantly and never updates.

Reply:
No 1!